apis/Google.Storage.V1/Google.Storage.V1/UpdateObjectOptions.cs

```diff
@@ -75,26 +86,38 @@ internal void ModifyRequest(UpdateRequest request)
             {
                 throw new ArgumentException($"Cannot specify {nameof(IfMetagenerationMatch)} and {nameof(IfMetagenerationNotMatch)} in the same options", "options");
             }
```

Comment:
Should there be another check up here that you if either of the metageneration pre-conditions are set, the IfGenerationMatch is also set? I'm not sure if that should always be the case, but it seems like it.